### DISCUSSION TITLE
Pinia standalone?

### DISCUSSION THREAD

**Discussion by andrei-gheorghiu** [vuejs/pinia#1183] [Help and Questions] [answered]
I know it's possible to use Pinia outside of a Vue project by actually adding vue (either v2 or v3) and creating an App solely for the purpose of starting the store (e.g: [reactivue](https://github.com/antfu/reactivue), [Using pinia standalone](https://github.com/vuejs/pinia/discussions/1051)).

Is there any plan to publish pinia really stand-alone (with a minimal list of dependencies handled internally - e.g: reactivity). 
I happen to believe it would take angular and react ecosystems by storm.

It would allow companies to encapsulate business logic in stores and use them across multiple frameworks, which no other state management plugin currently offers. It would be a huge selling point.

What I'm saying is you can't currently convince large corporate companies to import Vue and instantiate an app just for using pinia. They perceive as too heavy and they don't care it's not. But they would be willing to try a new state management "plugin".

**Comment by itsmnthn** [comment]
I would also love to use Pinia without depending on Vue app

**Comment by shtse8** [comment]
I want to use Pinia on server side program without vue framework.
the pattern is beautiful and I want to keep the same coding pattern even in my server side scripts.
is it possible?

```ts
import { createPinia, defineStore } from 'pinia'

const pinia = createPinia()

class Test {
  private name: string
  constructor(name: string) {
    this.name = name
    console.log(name)
  }

  run() {
    console.log(`${this.name}: run`)
  }
}

const useStore = defineStore('store', () => {
  return new Test('store')
})

console.log('start')
const store = useStore()
store.run()

```

**Comment by ilearnio** [comment]
Agree, it should not be depended on Vue. I switched on a new project that uses React, and wanted to use Pinia for it but it's too sad that it's tightly coupled with Vue (even though from conceptual perspective it shouldn't be)

**Comment by ColtHands** [comment]
Plus, with this one, would be great to switch between UI frameworks and keep the state

**Reply by u007** [reply]
yes right now the only alternative is to use nanostores from astrojs, but i would prefer pinia

**Comment by toyobayashi** [comment, marked as answer]
I created `pinia-core` package which replace the `vue-demi` with `@vue/runtime-core` and it actually works in React.

https://github.com/toyobayashi/pinia-core

**Reply by u007** [reply]
the rerender works with state change?

**Reply by toyobayashi** [reply]
yes, with the useSyncExternalStore, you can try to run the example code written in readme

**Reply by u007** [reply]
ah nice
